libs/plutus-preprocessor/src/Cardano/Ledger/Plutus/Preprocessor/Source/V3.hs

```diff
@@ -155,8 +157,8 @@ purposeIsWellformedWithDatumQ =
     purposeIsWellformedWithDatum arg =
       P.check $
         case unsafeFromBuiltinData arg of
-          PV3.ScriptContext txInfo _redeemer (PV3.SpendingScript txOutRef (Just _)) ->
-            not $ null $ P.filter ((txOutRef P.==) . PV3.txInInfoOutRef) $ PV3.txInfoInputs txInfo
+          PV3D.ScriptContext txInfo _redeemer (PV3D.SpendingScript txOutRef (Just _)) ->
+            not $ PLD.null $ PLD.filter ((txOutRef P.==) . PV3D.txInInfoOutRef) $ PV3D.txInfoInputs txInfo
```

Comment:
Would matching on the `txInfo` and the `txInInfoOutRef` improve the cost here? I remember that this had a large impact on the "newer" version of `AsData` (which we eventually discarded), but I can't remember if it has any effect on the current version of `AsData`.

Reply:
Perhaps, but is optimizing these scripts particularly important? @lehins

Reply:
NO, these are tests scripts, so they don't need to be fast. As long as they can fit within reasonable limits for per transaction execution limits we are fine